primitives/src/currency.rs

```diff
@@ -161,3 +161,10 @@ impl<
         Ok(asset.clone().into())
     }
 }
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
+pub enum CurrencyOrAsset {
+    NativeCurrency(TokenSymbol),
+    Asset(u32),
```

Comment:
Actually we could even do this
```suggestion
    Asset(AssetId),
```